app/controllers/nodes_controller.rb

```diff
@@ -0,0 +1,64 @@
+class NodesController < ApplicationController
+
+	def show
+		if params[:id]
+			@node = Node.find(params[:id])
+		else
+			@node = Node.root
+		end
+	end
+
+	def create
+		p = create_params
+		node = Node.create!(p)
+		redirect_to node_path(node.parent_id)
+	end
+
+	def edit
+		@node = Node.find(params[:id])
+	end
+
+	def update
+		@node = Node.find(params[:id])
+		if @node.update_attributes(node_params)
+			redirect_to node_path(@node.parent_id)
+      	else 
+      		flash[:error] = "Failed to update node"
+      		redirect_to edit_node_path(@node.id)
+		end
+	end
+
+	def destroy
+		@node = Node.find(params[:id])
+
+		if @node.children.any?
```

Comment:
Would it be better to use !@node.children.empty?, instead of @node.children.any?  Although it works in this case, any? is not checking that a list is empty, but that the values of the list are considered 'false' when converted to boolean, and for example  

 [false,false].any? 

is false, but the list [false,false] has 2 nodes. I would think that empty? gets better the semantic we want to check.